src/imap/session.rs

```diff
@@ -28,14 +29,32 @@ pub(crate) struct Session {
 pub(crate) trait SessionStream:
     tokio::io::AsyncRead + tokio::io::AsyncWrite + Unpin + Send + Sync + std::fmt::Debug
 {
+    // Make it possible to always change the read timeout on the session steram.
+    // We need to disable the timeout before going IDLE and re-enable it afterwards.
```

Comment:
I don't think it's clear what "always" means here, and it doesn't seem useful to comment where a function is used before 1. you can grep for that and 2. the comment quickly gets outdated.

```suggestion
    // Change the read timeout on the session stream.
    // This is done by recursively calling `set_inner_read_timeout()` on the
    // inner stream until we arrive at the `TimeoutStream` and can call
    // `set_read_timeout_pinned()`.
```

(typo: steram->stream)